addons/account/models/account_journal.py

```diff
@@ -654,9 +659,11 @@ def _fill_missing_values(self, vals):
             if journal_type in ('cash', 'bank') and not has_loss_account:
                 vals['loss_account_id'] = company.default_cash_difference_expense_account_id.id
 
-        # === Fill missing refund_sequence ===
+        # === Fill missing sequences ===
         if 'refund_sequence' not in vals:
             vals['refund_sequence'] = vals['type'] in ('sale', 'purchase')
+        if 'payment_sequence' not in vals:
+            vals['payment_sequence'] = vals['type'] in ('bank', 'cash')
```

Comment:
Instead, make `payment_sequence` a computed field dependent of `type` and remove this diff